subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/SpringBootPluginSmokeTest.groovy

```diff
@@ -85,6 +85,8 @@ class SpringBootPluginSmokeTest extends AbstractPluginValidatingSmokeTest implem
         smokeTestRunner.expectLegacyDeprecationWarning(BaseDeprecations.PROJECT_CONVENTION_DEPRECATION)
         smokeTestRunner.expectLegacyDeprecationWarning(BaseDeprecations.CONVENTION_TYPE_DEPRECATION)
         smokeTestRunner.expectLegacyDeprecationWarning(BaseDeprecations.JAVA_PLUGIN_CONVENTION_DEPRECATION)
+        smokeTestRunner.expectLegacyDeprecationWarning(BaseDeprecations.COPY_PROCESSING_SPEC_SET_FILE_MODE_DEPRECATION)
+        smokeTestRunner.expectLegacyDeprecationWarning(BaseDeprecations.FILE_TREE_ELEMENT_GET_MODE_DEPRECATION)
```

Comment:
Here we don't test the latest Spring Boot version.

* See https://github.com/gradle/gradle/issues/23937

But the latest will still be using the old API.

We should follow the same process as for Shadow.